### DISCUSSION TITLE
Node 22 - DeprecationWarning: The `punycode` module is deprecated.

### DISCUSSION THREAD

**Discussion by tebeco** [Azure/static-web-apps-cli#886] [General]
hello,

using the CLI is triggering this more and more:

```console
❯ node -v
v22.7.0

❯ npm -v
10.8.2

❯ npx @azure/static-web-apps-cli init

Welcome to Azure Static Web Apps CLI (2.0.1)

? Choose a configuration name: » bla(node:18152) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

❯ npm ls punnycode
+-- @azure/static-web-apps-cli@2.0.1
| `-- devcert@1.2.2
|   `-- is-valid-domain@0.1.6
|     `-- punycode@2.3.1 deduped
```

`devcert` needs to replace `is-valid-domain` since it's archived
I suppose Azure team could help it dependencies to be up to date as well
It would also help `SWA` CLI to be able to use Node22 for the build time and runtime to be ready in 2 month:
![image](https://github.com/user-attachments/assets/df27559d-da32-4bde-bf01-48a1f3ce1014)

**Comment by luckyabsoluter** [comment]
https://github.com/Azure/static-web-apps/issues/1534
